app/src/main/java/com/github/se/project/model/profile/Profile.kt

```diff
@@ -4,21 +4,70 @@ import java.util.EnumSet
 
 /** Data class representing a user profile. */
 data class Profile(
-    val uid: String, // Unique string id for the profile => TODO: link to google sign-in
-    val firstName: String, // First name of the user
-    val lastName: String, // Last name of the user
-    val phoneNumber: String, // Phone number of the user
-    val role: Role, // Role of the user
-    val section: Section, // Section of the user
-    val academicLevel: AcademicLevel, // Academic level of the user
-    // TODO: update languages and subjects to avoid using EnumSet
-    val languages: EnumSet<Language> = EnumSet.of(Language.STUDENT),
+    val uid: String,
+    val firstName: String,
+    val lastName: String,
+    val phoneNumber: String,
+    val role: Role,
+    val section: Section,
+    val academicLevel: AcademicLevel,
+    val languages: EnumSet<Language> = EnumSet.noneOf(Language::class.java),
     val subjects: EnumSet<TutoringSubject> = EnumSet.noneOf(TutoringSubject::class.java),
-    val schedule: List<List<Int>> = List(7) { List(12) { 0 } }, // Weekly schedule
-    var price: Int = 0,
-    // TODO: profile picture
+    val schedule: List<List<Int>> = List(7) { List(12) { 0 } },
+    var price: Int = 0
+    /*TODO: Add profile picture*/
 )
 
+data class ProfileUpload(
+    val uid: String,
+    val firstName: String,
+    val lastName: String,
+    val phoneNumber: String,
+    val role: String,
+    val section: String,
+    val academicLevel: String,
+    val languages: MutableList<String> = mutableListOf(),
+    val subjects: MutableList<String> = mutableListOf(),
+    val scheduleFlat: List<Int> = List(7 * 12) { 0 }, // Flattened version of weekly schedule
+    var price: Int = 0
+)
```

Comment:
The Profile and ProfileUpload classes contain similar fields but with slightly different data structures. You may want to consider using the same structure across both classes to avoid duplication. Consider adding a helper function or a mapper to streamline conversions and avoid redundant logic.

Reply:
I agree, we should be able to handle the EnumSet storing problem in Firestore by simply creating a helper function that converts the profile into a map :)